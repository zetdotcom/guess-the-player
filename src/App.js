import React from 'react';
import { useDispatch } from 'react-redux';
import './App.css';
import { fetchPlayers } from './features/players/playersSlice';
import PlayersContainer from './features/players/PlayersContainer';
import UserDetailsContainer from './features/userDetails/UserDetailsContainer';
import {GlobalStyle} from './style'
function App() {

  const dispatch = useDispatch();

  React.useEffect(() => {
    dispatch(fetchPlayers());
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  return (
    <div >
      <GlobalStyle />
      <PlayersContainer />
      <UserDetailsContainer />
    </div>
  );
}

export default App;
