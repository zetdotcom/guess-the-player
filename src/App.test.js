import React from 'react';
import { Provider } from 'react-redux';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
// import { initialState, reducer } from './reducer.js'
import store from './redux/store';
import App from './App';

afterEach(cleanup);

const renderComponent = newStore =>
	render(
		<Provider store={store(newStore)}>
			<App />
		</Provider>
	);
describe('TEST APP rendering', () => {
	test('renders App initial screen', () => {
		const { getByText, asFragment } = renderComponent();
		const header = getByText(/FanDuel Points Per Game/i);
		const radios = getByText(/Select how many players to show/);
		expect(header).toBeInTheDocument();
		expect(radios).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	test('should render App for winner screen', () => {
		// const state = store().getState();
		const winnerState = {
			// ...state,
			scoreboard: {
				// ...state.scoreboard,
				score: 10,
				attempts: 10,
				isWinner: true,
			},
		};
		const { getByText, asFragment } = renderComponent(winnerState);
		const won = getByText(/YOU WON/);
		expect(won).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});
});
