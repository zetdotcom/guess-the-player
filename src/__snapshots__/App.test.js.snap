// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST APP rendering renders App initial screen 1`] = `
<DocumentFragment>
  <div
    class="sc-AykKK emFRFx"
  >
    <h2>
      Guess which player has higher FanDuel Points Per Game (FPPG)
    </h2>
    <div>
      <div>
        <div>
          Select how many players to show:
        </div>
        <div
          class="sc-AykKG bKYsWi"
        >
          <label
            class="sc-AykKI chHbPs"
          >
            <input
              checked=""
              class="sc-AykKH dzTRW"
              name="players-number"
              type="radio"
              value="2"
            />
            2
          </label>
          <label
            class="sc-AykKI chHbPs"
          >
            <input
              class="sc-AykKH dzTRW"
              name="players-number"
              type="radio"
              value="3"
            />
            3
          </label>
          <label
            class="sc-AykKI chHbPs"
          >
            <input
              class="sc-AykKH dzTRW"
              name="players-number"
              type="radio"
              value="5"
            />
            5
          </label>
        </div>
        <div
          style="min-height: 25px; font-weight: lighter; padding: 5px; font-size: 18px;"
        />
        <ul
          class="sc-AykKJ gwAcfI"
        />
        <h4
          style="color: red; min-height: 25px;"
        />
        <button
          class="sc-AykKF kYUlLK"
        >
          NEXT
        </button>
      </div>
    </div>
    <div>
      <div
        data-testid="score"
      >
        Score: 0 / 10
      </div>
      <div
        data-testid="attempts"
      >
        Attempts: 0
      </div>
      <div
        data-testid="accuracy"
      >
        Accuracy: 
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TEST APP rendering should render App for winner screen 1`] = `
<DocumentFragment>
  <div
    class="sc-AykKK emFRFx"
  >
    <div>
      YOU WON
    </div>
    <div>
      <div
        data-testid="score"
      >
        Score: 10 / 10
      </div>
      <div
        data-testid="attempts"
      >
        Attempts: 10
      </div>
      <div
        data-testid="accuracy"
      >
        Accuracy: 100%
      </div>
    </div>
    <button
      class="sc-AykKF kYUlLK"
    >
       Play AGAIN 
    </button>
  </div>
</DocumentFragment>
`;
